StitchCore/Sources/StitchCore/Auth/Providers/UserPass/CoreUserPasswordAuthProviderClient.swift

```diff
@@ -1,25 +1,57 @@
+/**
+ * A class representing the routes on the Stitch server to perform various actions related to the username/password
+ * authentication provider.
+ */
 private final class Routes {
+    /**
+     * The authentication API routes that form the base of these provider routes.
+     */
     private let authRoutes: StitchAuthRoutes
+
+    /**
+     * The name of the provider for these routes.
+     */
     private let providerName: String
 
+    /**
+     * Initializes the routes with a `StitchAuthRoutes` and the name of the provider.
+     */
     fileprivate init(withAuthRoutes authRoutes: StitchAuthRoutes,
                      withProviderName providerName: String) {
         self.authRoutes = authRoutes
         self.providerName = providerName
     }
 
-    private func extensionRoute(forPath path: String) -> String{
+    /**
+     * Private helper which returns the provided path, appended to the base route for the authentication provider.
+     */
+    private func extensionRoute(forPath path: String) -> String {
         return "\(authRoutes.authProviderLoginRoute(withProviderName: providerName))/\(path)"
```

Comment:
[minor] Not code you wrote but I'm pretty sure this has to be wrong. The extension comes after the auth provider name, not `/login`.

Reply:
@jsflax, this is where the spec would come into handy if it defined literally the URL format that must match.

Reply:
We should write a test for this. I believe it should be `authProviderRoute`, not `authProviderLoginRoute`, and that should clean this up.